src/descformats/base.py

```diff
@@ -0,0 +1,284 @@
+        pass
+
+
+class QPHandle(DataHandle):
+    """DataHandle for qp ensembles
+    """
+    suffix = 'hdf5'
+
+    @classmethod
+    def _open(cls, path, mode, **kwargs):
+        """Open and return the associated file
+        Notes
+        -----
+        This will simply open the file and return a file-like object to the caller.
+        It will not read or cache the data
+        """
+        return tables_io.io.io_open(path, mode=mode, **kwargs)  #pylint: disable=no-member
+
+    @classmethod
+    def _read(cls, path, **kwargs):
+        """Read and return the data from the associated file """
+        return qp.read(path)
+
+    @classmethod
+    def _write(cls, data, path, **kwargs):
+        """Write the data to the associatied file """
+        return data.write_to(path)
+
+    @classmethod
+    def _initialize_write(cls, data, path, data_lenght, **kwargs):
+        comm = kwargs.get('communicator', None)
+        return data.initializeHdf5Write(path, data_lenght, comm)
+
+    @classmethod
+    def _write_chunk(cls, data, fileObj, groups, start, end, **kwargs):
+        return data.writeHdf5Chunk(fileObj, start, end)
+
+    @classmethod
+    def _finalize_write(cls, data, fileObj, **kwargs):
+        return data.finalizeHdf5Write(fileObj)
+
+
+HDFFile = Hdf5Handle
+
+
+FitsFile = FitsHandle
+
+
+class TextFile(DataFile):
+    """
+    A data file in plain text format.
+    """
+    suffix = 'txt'
+    format = "http://edamontology.org/format_2330"
+    
+    @classmethod
+    def _read(cls, path, **kwargs):
+        """Read and return the data from the associated file """
+        with cls._open(path, mode='r') as fin:
+            return fin.read()
+
+    @classmethod
+    def _write(cls, data, path, **kwargs):
+        """Write the data to the associatied file """
+        with cls._open(path, mode='w') as fout:
+            fout.write(data)
+
+
+class YamlFile(DataFile):
+    """
+    A data file in yaml format.
+    """
+    suffix = 'yml'
+    format = "http://edamontology.org/format_3750"
+
+    @classmethod
+    def _read(cls, path, **kwargs):
+        """Read and return the data from the associated file """
+        with cls._open(path, mode='r') as fin:
+            return yaml.load(fin, Loader=Loader)
+
+    @classmethod
+    def _write(cls, data, path, **kwargs):
+        """Write the data to the associatied file """
+        with cls._open(path, mode='w') as fout:
+            yaml.dump(data, fout, Dumper=Dumper)
+
+
+class Directory(DataFile):
+    suffix = ''
+
+    @classmethod
+    def _open(cls, path, mode, **kwargs):
+        p = pathlib.Path(path)
+
+        if mode == "w":
+            if p.exists():
+                shutil.rmtree(p)
+            p.mkdir(parents=True)
+        else:
+            if not p.is_dir():
+                raise ValueError(f"Directory input {path} does not exist")
+        return p
+
+    @classmethod
+    def _close(cls, fileObj, **kwargs):
+        pass
```

Comment:
There's a few TX base types not included here, like FileCollection, PickleFile, PNGFile. Is that deliberate?

Reply:
No, it is probably that they got lost when I split stuff between base and tx/tx.py